svelte-responsive-layout/src/TopNav.svelte

```diff
@@ -0,0 +1,122 @@
+<script>
+  import ToggleIcon from './icons/ToggleIcon.svelte';
+
+  export let items;
+
+  let isActive = false;
+
+  function handleToggleMenu() {
+    isActive = !isActive;    
+  }
+
+  function handleCloseToggleMenu() {
+    isActive = false;    
+  }
+</script>
+
+<style type="text/postcss">
+  .top-nav {
+    color: var(--secondary-color);
+
+    &-toggler {
+      display: flex;
+      align-items: center;
+      height: 24px;
+      width: 24px;
+      padding: 0;
+      border: none;
+      background-color: transparent;
+
+      @media(--big-viewport) {
+        display: none;
+      }
+    }
+
+    &-list {
+      display: flex;
+      list-style-type: none;
+      position: fixed;
+      overflow: auto;
+      overscroll-behavior: contain;
+      top: 0;
+      bottom: 0;
+      left: 0;
+      right: 0;
+      background-color: rgba(0, 0, 0, 0.9);
+      opacity: 0;
+      pointer-events: none;
+      flex-direction: column;
+      align-items: center;
+      margin-top: var(--header-height);
+      padding: 1em;
+      color: var(--main-color);
+      font-size: 2em;
+
+      @media(--big-viewport) {
+        position: static;
+        background-color: transparent;
+        color: inherit;
+        font-size: 1em;
+        flex-direction: row;
+        margin: 0;
+        padding: 0;
+        opacity: 1;
+        pointer-events: auto;
+      }
+
+      .active & {
+        opacity: 1;
+        pointer-events: auto;
+        transition: opacity 0.3s;
+      }
+
+      a {
+        color: inherit;
+        opacity: 0.6;
+        text-decoration: none;
+        transition: opacity 0.3s;
+        
+        &:hover {
+          opacity: 1;
+          text-decoration: underline;
+        }
+      }
+
+      & > li {
+        &:not(:last-child) {
+          margin-bottom: 0.5em;
+        }
+
+        @media(--big-viewport) {
+          font-size: 0.7em;
+
+          &:not(:last-child) {
+            margin-bottom: 0;
+            margin-right: 1em;
+          }
+        }
+      }            
+    }
+  }
+</style>
+
+<nav
+  class="top-nav"
+  class:active={isActive}
+>
+  <button
+    class="top-nav-toggler"
+    on:click={handleToggleMenu}
+  >
+    <ToggleIcon isActive={isActive} />
+  </button>
```

Comment:
Let's do an A11Y (Accessibility) exercise here: Try to navigate this page using the keyboard only. Use Tab to go forward and Shift + Tab to go backward.

On mobile, you'll notice that while the menu is closed, you still have access to the links. My challenge here to you is:
When the menu is closed, it goes from the burger button directly to "Discover Svelte". Only if the menu is open, it goes through the menu links. There are a lot of articles here and there about this, but for this, I think the best example is this talk from Sara Soueidan - https://vimeo.com/331530115 - The video is full of a11y tips. For this particular scenario, watch from ~18:15 to ~22:45.